roxie/ccd/ccdstate.cpp

```diff
@@ -918,18 +918,23 @@ class CRoxieQuerySetManager : public CInterface, implements IRoxieQuerySetManage
     unsigned channelNo;
     bool active;
     StringAttr querySetName;
+    CriticalSection crit;   // For parallel load
 
     void addQuery(const char *id, IQueryFactory *n, hash64_t &hash)
     {
         hash = rtlHash64Data(sizeof(hash), &hash, n->queryHash());
-        queries.setValue(id, n);
+        {
+            CriticalBlock b(crit);
+            queries.setValue(id, n);
+        }
         n->Release();  // setValue links
     }
 
     void addAlias(const char *alias, const char *original, hash64_t &hash)
     {
         if (original && alias)
         {
+            CriticalBlock b(crit);
```

Comment:
This could move before the setValue (since queries should be constant at this point), which might make executing these in parallel more useful.